File: samples/OCaml/primality-by-trial-division.ml
let is_prime n =
  if n = 2 then true
  else if n < 2 || n mod 2 = 0 then false
  else
    let rec loop k =
      if k * k > n then true
      else if n mod k = 0 then false
      else loop (k+2)
    in loop 3

File: samples/Mathematica/primality-by-trial-division.m
IsPrime[n_Integer] :=
 Module[{k = 2},
  If[n <= 1, Return False];
  If[n == 2, Return True];
  While[k <= Sqrt[n],
   If[Mod[n, k] == 0, Return[False], k++]
   ];
  Return[True]
 ]
